wasm/wasm_source/src/tx_bridge_pool.rs

```diff
@@ -1,25 +1,61 @@
 //! A tx for adding a transfer request across the Ethereum bridge
 //! into the bridge pool.
-use std::collections::HashSet;
-
 use borsh::{BorshDeserialize, BorshSerialize};
-use eth_bridge_pool::{GasFee, PendingTransfer};
+use eth_bridge::storage::{bridge_pool, native_erc20_key, wrapped_erc20s};
+use eth_bridge_pool::{GasFee, PendingTransfer, TransferToEthereum};
 use namada_tx_prelude::*;
 
 #[transaction]
-fn apply_tx(tx_data: Vec<u8>) {
+fn apply_tx(ctx: &mut Ctx, tx_data: Vec<u8>) -> TxResult {
     let signed = SignedTxData::try_from_slice(&tx_data[..]).unwrap();
-    let transfer =  PendingTransfer::try_from_slice(
-        &signed.data.unwrap()[..]
-    )
-    .unwrap();
+    let transfer =
+        PendingTransfer::try_from_slice(&signed.data.unwrap()[..]).unwrap();
     // pay the gas fees
-    let GasFee {
+    let GasFee { amount, ref payer } = transfer.gas_fee;
+    token::transfer(
+        ctx,
+        payer,
+        &bridge_pool::BRIDGE_POOL_ADDRESS,
+        &address::xan(),
+        None,
+        amount,
+    )?;
+    let TransferToEthereum {
+        ref asset,
+        ref sender,
         amount,
-        ref payer,
-    } = transfer.gas_fees;
-    token::transfer(payer, &BRIDGE_POOL_ADDRESS, &address::xan(), amount);
+        ..
+    } = transfer.transfer;
+    // if minting wNam, escrow the correct amount
+    if *asset == native_erc20_address(ctx) {
```

Comment:
nit: does this `asset` need a `ref` here? I don't see it being used anywhere else, so maybe you could just do the comparison: `asset == native_erc20_address(ctx)`